File: employee-portal/src/services/employeeAuthService.ts
import { employeeApiClient } from "./apiClient";
import type { EmployeeLoginRequest, EmployeeLoginResponse, EmployeeResponse } from "@/types/employee";

export const employeeAuthService = {
  async login(request: EmployeeLoginRequest): Promise<EmployeeLoginResponse> {
    const { data } = await employeeApiClient.post<EmployeeLoginResponse>("/api/v1/employees/auth/login", request);
    return data;
  },

  async getCurrentEmployee(): Promise<EmployeeResponse> {
    const { data } = await employeeApiClient.get<EmployeeResponse>("/api/v1/employees/me");
    return data;
  },
};

File: backend/services/employee-service/src/main/java/com/banksphere/employee/service/EmployeeAuthService.java
package com.banksphere.employee.service;

import com.banksphere.employee.dto.EmployeeLoginRequest;
import com.banksphere.employee.dto.EmployeeLoginResponse;
import com.banksphere.employee.dto.EmployeeResponse;

import java.util.UUID;

public interface EmployeeAuthService {

    EmployeeLoginResponse login(EmployeeLoginRequest request);

    EmployeeResponse getCurrentEmployee(UUID employeeId);
}
